test/sqllogictest/joins.slt

```diff
@@ -250,3 +250,19 @@ SELECT * FROM
 ----
 baz  foo  bar
 1    1    1
+
+# Test that join planning with constants does not panic due to missing arrangements.
```

Comment:
hmm, looking at the current plan for this query:
```
%0 =
| Get materialize.public.l2 (u21)
| ArrangeBy ()

%1 =
| Constant (1) (2)

%2 =
| Constant (2) (3)

%3 =
| Join %0 %1 %2 (= #2 #3)
| | implementation = Differential %1 %2.(#0) %0.()
| | demand = (#0..#2)
| Map 2, 3
| Project (#0..#2, #4, #2, #5)
```
My impression is that it's not so much a defect in join planning so much as:
1) we're missing the feature whereby `FoldConstants` collapses a join containing 2+ constant inputs into a join containing max 1 constant RelationExpr (there's already a TODO in `FoldConstants`)
2) a cross join on a constant should become a `FlatMap` instead.

Reply:
This example just got chosen because it was the easiest way to trip the error that was about to show up in @benesch's lateral join work. There, you can force the constant expression to not be the first argument to a join (by making it a lateral join with a constant collection) but otherwise forcing the constant collection second (which is where it would cause the error) required this sort of silly query.

The query can also be planned better, and if it does we should revisit whether this test actually tests what it intends to test.